src/main/java/com/yahoo/bullet/parsing/LeafExpression.java

```diff
@@ -40,9 +44,15 @@ public String toString() {
 
     @Override
     public Optional<List<BulletError>> initialize() {
-        if (value == null || value.getKind() == Value.Kind.VALUE && !Type.PRIMITIVES.contains(value.getType())) {
+        if (value == null) {
+            return Optional.of(Collections.singletonList(LEAF_EXPRESSION_REQUIRES_VALUE_FILED_ERROR));
+        }
+        if (value.getType() != null && !Type.PRIMITIVES.contains(value.getType())) {
             return Optional.of(Collections.singletonList(LEAF_EXPRESSION_REQUIRES_PRIMITIVE_TYPE_ERROR));
         }
+        if (value.getKind() == Value.Kind.VALUE && value.getType() == null) {
```

Comment:
What if kind is null? I guess this matters for BinaryExpression too

Reply:
Done.